### DISCUSSION TITLE
pgb2file_f024_24_0p25: No such file or directory

### DISCUSSION THREAD

**Discussion by benjamin-cash** [NOAA-EMC/global-workflow#3069] [Q&A]
I am hitting an error in atmos_prod where it lists the final hour (so f024_24, or f048_48) as no such file or directory. I'm afraid I'm not familiar enough with the prod tasks to understand where this is going wrong and was hoping someone could help me out. The log file from one of the failed tasks is attached here. Thanks!
[atmos_prod_mem001_f024.log](https://github.com/user-attachments/files/17636036/atmos_prod_mem001_f024.log)

**Comment by DavidHuber-NOAA** [comment]
I suspect this is an instance of regenerating/rerunning an experiment after a failed forecast.  The forecast generates restart files in the `${DATAROOT}/gefs.YYYYMMDDHH` directory.  If it fails, those files stick around so that the forecast can pick up from where it left off.  But if you start out the `COMROOT` directory, then the `atmos_prod` jobs will not have the files it needs.

I suggest starting fresh by deleting `DATAROOT` (`/scratch1/02441/bcash/stmp//RUNDIRS/C192mx025_nonans`) and `COMROOT` (`/scratch1/02441/bcash/COMROOT`), then starting the experiment over.

On a side note, I think we should add a feature to the `setup_expt.py` script to check if `DATAROOT` exists and ask users if they would like to delete it.

**Reply by benjamin-cash** [reply]
Hi @DavidHuber-NOAA - good to know! This brings up a bit of a general question - as far as I can tell, the model runs all completed successfully, but I did have multiple failed atmos_prod jobs as I did some debugging. This was just a 48h test run, so rerunning everything clean isn't much an issue, but in the future I will be making much longer runs that will be expensive to repeat. Is there a way to simply remove the files that atmos_prod is choking on without rerunning the entire experiment?

**Reply by DavidHuber-NOAA** [reply]
@benjamin-cash Unfortunately not.  The issue is that the forecast model starts at a later forecast hour (if it finds a restart file at 36 hours, that's where it will start).  But if you restarted the experiment, then the COMROOT directory will not contain the forecast products for hours 0-33, which is what causes the `prod` jobs to fail.  The only way to regenerate those files is to clear out `DATAROOT` and restart the forecast so it will start at forecast hour 0.

**Reply by benjamin-cash** [reply]
Oof. Well, not the answer I wanted, but much appreciated so that I don't spend time banging my head against this - many thanks. :)

**Reply by benjamin-cash** [reply]
Actually, one more question for you. In my case, I didn't realize that this was an issue until the forecast ensemble had completed and a large number of prod tasks had been scheduled. Is there an efficient way to rewind rocoto to the forecast tasks, or do I literally have to go through and rewind every individual prod task? Somewhat baffling, not even deleting the .db file has caused rocoto to reset to the beginning. I would really like to avoid having to create a new experiment directory and make all of the necessary changes to my .xml file again if I can avoid it...

**Reply by DavidHuber-NOAA** [reply]
Yes, you can rewind by targeting the forecast metatask.  The metatasks are still grouped by member, but at least you don't have to rewind all of the segments.  You can probably do it in a loop as well.  Something like

```bash
num_members=  # set the member count
cycle=1994050100
for mem in $(seq 0 $num_mems); do
   mem3=$(printf %03d $mem)
   metatask="gefs_fcst_mem${mem3}"
   rocotorewind -w C192mx025_nonans.xml -d C192mx025_nonans.db -c $cycle -m $metatask
done
```

**Comment by benjamin-cash** [comment]
Hi @DavidHuber-NOAA - I made a new clean run where the fcst task ran and exited without issue, and f000 ran without issue for atmos_prod but f024 crashed with the same error. I've attached the various log files here in case something else suggests itself.
[fcst_mem002_seg0.log](https://github.com/user-attachments/files/17682439/fcst_mem002_seg0.log)
[atmos_prod_mem002_f024.log](https://github.com/user-attachments/files/17682440/atmos_prod_mem002_f024.log)
[atmos_prod_mem002_f000.log](https://github.com/user-attachments/files/17682441/atmos_prod_mem002_f000.log)

**Reply by DavidHuber-NOAA** [reply]
@aerorahul It looks like the offending issue is noted on line 7637 of [atmos_prod_mem002_f024.log](https://github.com/user-attachments/files/17682440/atmos_prod_mem002_f024.log):

```
*** FATAL ERROR: grib message #48 not found for tmpfile_f024 ***
```

Later, the job fails because it cannot find `pgb2file_f024_24_0p25`.  Is this just an issue of needing to query how many messages are in a GRIB2 file?  Or should there always be 48 messages and this indicates an issue with the UPP?

**Reply by aerorahul** [reply]
I am not sure about this.  @WenMeng-NOAA may know.

**Reply by WenMeng-NOAA** [reply]
@DavidHuber-NOAA @aerorahul The original exglobal_atmos_products.sh script was designed for GFS product generation.  The UPP output for GFS includes more than 1100 fields. MPMD (24 tasks) has been used to speed up interpolation from the gaussian grid into the lat-lon grid. The UPP output from GEFS should have fewer fields than GFS (might be 600 fields). I would suggest reducing MPMD for GEFS to fewer than 24.
```
In atmos_prod_mem002_f024.log, 
nproc=24
ncount=48

Compare with the process for GFS:
nproc=24
ncount=751
```

**Reply by benjamin-cash** [reply]
@NeilBarton-NOAA @XiaqiongZhou-NOAA - tagging you in for your awareness.

**Reply by aerorahul** [reply]
Not sure why `ncount=48` in the `atmos_prod_mem002_f024.log` file.  Is GEFS/SFS running with a significantly reduced field parm table?

**Reply by WenMeng-NOAA** [reply]
https://github.com/NOAA-EMC/global-workflow/blob/develop/parm/product/sfs.0p25.fFFF.paramlist.a.txt
Only 52 fields are retrieved from the UPP output for SFS. One task should be good for processing.

**Reply by aerorahul** [reply]
That's it.  Another reason why we need a `parm/config/sfs` space.

**Comment by aerorahul** [comment]
@benjamin-cash 
Is this discussion relevant or can it be closed?

**Reply by benjamin-cash** [reply]
It can be closed - now that the prod tasks have been removed for SFS it has been overtaken by events.
